src/tiny_slam/tiny_fascade.h

```diff
@@ -9,20 +9,45 @@
 
 #include "tiny_world.h"
 
+/*!
+ * \brief Derived class from LaserScanObserver to manage all world
+ *
+ * Handles data came from laser scanner and locates views if he is set.
+ * Presents a pattern fascade for all algorithm "tiny slam"
+ *
+ */
 class TinySlamFascade : public LaserScanObserver {
 private:
   using ScanMatcherObsPtr = std::shared_ptr<GridScanMatcherObserver>;
 public: // methods
   // TODO: copy ctor, move ctor, dtor
+
+  /*!
+   * Parameterized constructor sets all data members
+   * \param[in] gcs           - Grid Cell Strategy - configuration of cells in map
+   * \param[in] params        - configuration of laser scan (sets the quality value)
+   * \param[in] skip_max_vals - boolean variable - flag shows how are high values in messages will be handeled
+   */
   TinySlamFascade(std::shared_ptr<GridCellStrategy> gcs,
                   const TinyWorldParams &params,
                   bool skip_max_vals):
     LaserScanObserver(skip_max_vals), _world(new TinyWorld(gcs, params)) {}
 
+  /*!
+   * Setter for data member "viewer"
+   * \param[in] viewer - new value of data member viewer
+   */
   void set_viewer(std::shared_ptr<RvizGridViewer> viewer) {
     _viewer = viewer;
   }
 
+  /*!
+   * Updates the building map and robot pose using scan data
+   * 
+   * It shifts the robot pose on scanner odnometry parameters (\f$\Delta x, \Delta y, \Delta\theta\f$) and update the map using abilities presented in "world"
```

Comment:
The update is done by prediction-correction approach (odomentry is used for prediction, laser scan -- for correction).